Service/Manager.php

```diff
@@ -321,6 +332,38 @@ public function refresh(array $params = [])
     }
 
     /**
+     * Adds ids to documents
+     *
+     * @param array $bulkQueries
+     *
+     * @param array $bulkResponse
+     */
+    public function addIds(array $bulkQueries, $bulkResponse = [])
+    {
+        if (empty($bulkResponse)) {
+            $this->persistedObjects = [];
+            return;
+        }
+        $indexing = [];
+        foreach ($bulkQueries['body'] as $number => $query) {
+            if (isset($query['index']) && !isset($query['index']['_id'])) {
+                $indexing[] = $number / 2;
+            }
+        }
+
+        if (isset($bulkQueries['body'][0]['index'])) {
+            if (isset($this->persistedObjects)) {
+                $i = 0;
+                foreach ($this->persistedObjects as $document) {
+                    $document->id = $bulkResponse['items'][$indexing[$i]]['create']['_id'];
```

Comment:
You can not hardcode `id` property here. User is free to choose which property of a document he wants to use for Elasticsearch `_id` value (using `@Id` annotation).

You could get actual property from metadata collector.